.github/workflows/build-frequent.yaml

```diff
@@ -124,11 +124,80 @@ jobs:
       gcchash: ${{ needs.init-submodules.outputs.gcchash }}
       multilib: ${{ matrix.multilib }}
 
+  compare-artifacts:
+    needs: [creg]
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Download and compare
+        run: |
+          pip install pygithub requests
+          mkdir logs
+          mkdir summaries
+          python ./scripts/download_artifacts.py -hash ${{ needs.init-submodules.outputs.gcchash }} -token ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Make artifact zips
+        run: |
+          zip -r summaries.zip summaries
+          zip -r logs.zip logs
+
+      - name: Upload compare summaries
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ needs.init-submodules.outputs.gcchash }}-summaries
+          path: |
+            summaries.zip
+          retention-days: 5
+
+      - name: Upload log failures
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ needs.init-submodules.outputs.gcchash }}-logs
+          path: |
+            logs.zip
+          retention-days: 5
+
   generate-issues:
     if: always() # Generate github issues even when some (or all) targets fail to build
     runs-on: ubuntu-22.04
-    needs: [creg]
+    needs: [compare-artifacts]
+    permissions:
+      issues: write
     steps:
-      - name: Generate summary
+      - uses: actions/checkout@v3
+
+      - name: Download summaries artifact
+        id: download-summaries-artifact
```

Comment:
This is just a nitpick so feel free to ignore, but the ID isn't needed unless you're using the step's output or need to refer to it in another step.